src/server.rs

```diff
@@ -44,16 +42,29 @@ impl Server {
         let router = self.router.clone();
         let headers = self.headers.clone();
 
+        let asyncio = py.import("asyncio").unwrap();
+
+        let event_loop = asyncio.call_method0("new_event_loop").unwrap();
```

Comment:
are we sure this is on a python thread? (that will keep executing the event loop with the control on python's side, before we start throwing tasks into it)

Reply:
Oh nevermind, i just saw the `run_forever` pass to python and then asyncio down below, yeah, that should make this alright.